Oauth metrics docs (#7855)
strimzi/strimzi-kafka-operator: documentation/assemblies/metrics/assembly-metrics.adoc

You can monitor your Strimzi deployment by viewing key metrics on dashboards and setting up alerts that trigger under certain conditions.
Metrics are available for each of the components of Strimzi.

You can also collect metrics specific to `oauth` authentication and `opa` or `keycloak` authorization. 

> You do this by setting the `enableMetrics' property to `true` in the listener configuration of the `Kafka` resource. 

Author: ```suggestion
You do this by setting the `enableMetrics` property to `true` in the listener configuration or authorization configuration of the `Kafka` resource.
```

Author: @PaulRMellor Also fixed a backtick there.

You can monitor your Strimzi deployment by viewing key metrics on dashboards and setting up alerts that trigger under certain conditions.
Metrics are available for each of the components of Strimzi.

> If you are using `oauth` authentication or `opa` / `keycloak` authorization the collection of their specific metrics has to be explicitly enabled. For example, Kafka CR 'spec.kafka.listeners.authentication' and 'spec.kafka.authorization' have 'enableMetrics' property which needs to be set to 'true'.

Reviewer: ```suggestion
You can also collect metrics specific to `oauth` authentication and `opa` or `keycloak` authorization. 
You do this by setting the `enableMetrics' property to `true` in the listener configuration of the `Kafka` resource. 
For example, set `enableMetrics'  to `true` in `spec.kafka.listeners.authentication` and `spec.kafka.authorization`.
```

Author: Maybe to be consistent:

You do this by setting the `enableMetrics' property to `true` in the listener configuration or authorization configuration of the `Kafka` resource.

> Similarly for client authentication in KafkaBridge, KafkaConnect, KafkaMirrorMaker, or KafkaMirrorMaker2 CRs when 'oauth' type is used.

Reviewer: ```suggestion
Similarly, you can enable metrics for `oauth` authentication in the `KafkaBridge`, `KafkaConnect`, `KafkaMirrorMaker`, and `KafkaMirrorMaker2` custom resources.  
```

> Similarly for client authentication in KafkaBridge, KafkaConnect, KafkaMirrorMaker, or KafkaMirrorMaker2 CRs when 'oauth' type is used.

Reviewer 2: I think this is misleading as it suggests this applies in general. While in reality, it applies only to `oauth` authentication and `opa` authorization / `keycloak` authentication. So I think it needs to be approached from the other side ... i.e. _if you are using one of these, you will need to explicitly enable the metrics ..._

Author: Ok, how about:

```suggestion
If you are using `oauth` authentication or `opa` / `keycloak` authorization the collection of their specific metrics has to be explicitly enabled. For example, Kafka CR 'spec.kafka.listeners.authentication' and 'spec.kafka.authorization' have 'enableMetrics' property which needs to be set to 'true'.
Similarly for client authentication in KafkaBridge, KafkaConnect, KafkaMirrorMaker, or KafkaMirrorMaker2 CRs when 'oauth' type is used.
```

Reviewer 2: Yeah, I think this sounds good to me. It makes it more clear who should worry about this.
